sample-authz/src/main/java/net/trajano/ms/example/authz/HttpAuthorizationHeaders.java

```diff
@@ -0,0 +1,51 @@
+package net.trajano.ms.example.authz;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * Utility class to process HTTP Headers dealing with authorization.
+ */
+public final class HttpAuthorizationHeaders {
+
+    /**
+     * Basic Authorization pattern. It is "Basic " followed by a valid base64
+     * string. The Base64 pattern was from
+     * https://stackoverflow.com/a/8106083/242042
+     */
+    private static final Pattern BASIC_AUTHORIZATION_PATTERN = Pattern.compile("^Basic ((?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?)$");
+
+    /**
+     * This will extract the username and password components of a basic
+     * authorization header. If there is any problem this will return null.
+     *
+     * @param authorization
+     *            authorization
+     * @return an array consisting of the user name and password as strings.
+     */
+    public static String[] parseBasicAuthorization(@NotNull final String authorization) {
+
+        if (authorization == null) {
+            return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty array instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1168)